File: backend/app/models/topology.py
"""
拓扑图版本 SQLAlchemy 模型
"""
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base, utc_now


class Topology(Base):
    __tablename__ = "topologies"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    version_no: Mapped[str] = mapped_column(String(64), unique=True, nullable=False)
    title: Mapped[str | None] = mapped_column(String(255), nullable=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    drawio_xml: Mapped[str] = mapped_column(Text, nullable=False)
    created_by: Mapped[int | None] = mapped_column(
        Integer, ForeignKey("users.id"), nullable=True
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=utc_now
    )
    is_current: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)

    def __repr__(self) -> str:
        return f"<Topology id={self.id} version={self.version_no}>"

File: frontend/src/types/topology.ts
/**
 * 拓扑图 TypeScript 类型
 */

export interface TopologyVersion {
  id: number
  version_no: string
  title: string | null
  created_at: string
  is_current: boolean
}

export interface TopologyDetail {
  id: number
  version_no: string
  title: string | null
  description: string | null
  drawio_xml: string
  created_by: number | null
  created_at: string
  is_current: boolean
}

export interface TopologyGenerateRequest {
  provider?: string
  model?: string
}

export interface TopologySaveRequest {
  title?: string
  description?: string
  drawio_xml: string
}
